### DISCUSSION TITLE
Should we handle category URL generation from the code rather than database?

### DISCUSSION THREAD

**Discussion by devansh-webkul** [bagisto/bagisto#5142] [General]
## Discussion
I am just raising this to know the point of view of this because the current URL generation is happening from the triggers and function. I know that handling the URLs from SQL is robust but it is creating problems for most of the users.

## Problems
- When I try to import and export the data then sometimes trigger functions not came which results in an error.
- Explicitly defining the definers and host for the URL generations.
- Not having good support in other platforms like Postgres, etc.

## Related Issues
- https://github.com/bagisto/bagisto/issues/4368
- https://github.com/bagisto/bagisto/issues/3318

**Comment by clain4u** [comment]
Automatic URL generation triggered by database should be avoided in any case due to following reasons.

1. Any logic should be within the code and not in the database.
2. Your logic of URL generation may not apply to the user
3. Effects manual update of URL's for bulk updates.
4. Users are left with no clue on where this is generated form.

Personally, I find this problematic while importing large set of categories using automated DB insert or a sync.

**Comment by devansh-webkul** [comment]
Fixed all the issues in major release, so closing this now.
